frontend/pages/challenges/questionnaire.js

```diff
@@ -1,22 +1,204 @@
+import { useState } from "react";
+import styles from "./questionnaire.module.css";
+
+function AddQuestion({ onSubmit }) {
+  const [question, setQuestion] = useState();
+  const [isMultiple, setIsMultiple] = useState(false);
+  const [options, setOptions] = useState([{ value: "", isCorrect: false }]);
+
+  const removeOption = (index) => {
+    setOptions((prevOptions) => prevOptions.filter((_, i) => i !== index));
+  };
+
+  const disabled = options.every(({ isCorrect }) => !isCorrect) || !question;
+
+  const handleSubmit = () => {
+    if (disabled) return;
+    onSubmit({
+      question,
+      isMultiple,
+      options,
+    });
+    setQuestion("");
+    setIsMultiple(false);
+    setOptions([{ value: "", isCorrect: false }]);
+  };
+
+  return (
+    <div className={styles.AddQuestion}>
+      <h4>Question</h4>
+      <input
+        type="text"
+        value={question}
+        onChange={({ target }) => setQuestion(target.value)}
+      />
+      <div>
+        <h4>Answers</h4>
+        <p>Write options and check the ones that are correct</p>
+        <div>
+          <input
+            type="checkbox"
+            checked={isMultiple}
+            onChange={() => {
+              setOptions((prev) =>
+                prev.map((o) => ({ ...o, isCorrect: false }))
+              );
+              setIsMultiple((p) => !p);
+            }}
+          />
+          <label>There is more than one correct answer</label>
+        </div>
+      </div>
+
+      <div className={styles.options}>
+        {options.map((option, index) => (
+          <div className={styles.option} key={`${option}-${index}`}>
+            <input
+              type="text"
+              value={option.value}
+              onChange={({ target }) => {
+                const optionsCopy = [...options];
+                optionsCopy[index].value = target.value;
+                setOptions(optionsCopy);
+              }}
+            />
+            <button
+              className={styles.secondary}
+              onClick={() => removeOption(index)}
+            >
+              Delete
+            </button>
+            <input
+              checked={option.isCorrect}
+              onChange={() => {
+                const optionsCopy = [...options];
```

Comment:
Too much logic in the render function. This is an antipattern as it makes it harder to read, as well as add additional helpers like memoing, shared code, etc.